context/app/static/js/components/detailPage/provenance/utils.ts

```diff
@@ -21,11 +42,35 @@ export function nonDestructiveMerge<T extends Record<string, unknown>, U extends
 ): T {
   const merged = { ...a } as Record<string, unknown>;
```

Comment:
Can we provide stricter types for the provenance data?

Reply:
The types are currently provided via the generics, i.e. this function gets called with `ProvData` objects, then gets cast back to the ProvData type at the end with the `return merged as T`. Prior to adding this most recent prov-only logic, this was a more generic method we could reuse elsewhere; if we end up needing a similar functionality in the future, we could generify this approach by extracting the prov-specific logic into a function param.